core/domain/skill_jobs_one_off.py

```diff
@@ -116,3 +116,63 @@ def reduce(key, values):
                 sum(ast.literal_eval(v) for v in values))])
         else:
             yield (key, values)
+
+
+class SkillCommitCmdMigrationOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """This job is used to migrate the old commit cmds in skill commit log
+    model to the latest cmd format.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [skill_models.SkillCommitLogEntryModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        updated_commit_cmds = []
+        update_required = False
+        for commit_cmd_dict in item.commit_cmds:
+            updated_commit_cmd_dict = {}
+            for cmd_key, cmd_val in commit_cmd_dict.items():
+                if cmd_key == 'explanation':
```

Comment:
Why this special check? Also is `updated_commit_cmds` only used if this `if` block is called (due to setting of `update_required`)?

Reply:
This is the only old command which we found during prod validation, so that is why we just do this check via the `if`. And yes, `updated_commit_cmds` is only used if the `if` block is called